pkg/handlers/clusterlogging/accesscontrol/documentmanager.go

```diff
@@ -0,0 +1,102 @@
+package accesscontrol
+
+import (
+	"time"
+
+	"github.com/openshift/elasticsearch-proxy/pkg/apis/security"
+	"github.com/openshift/elasticsearch-proxy/pkg/clients"
+	"github.com/openshift/elasticsearch-proxy/pkg/config"
+	cl "github.com/openshift/elasticsearch-proxy/pkg/handlers/clusterlogging/types"
+	log "github.com/sirupsen/logrus"
+)
+
+//DocumentManager understands how to load and sync ACL documents
+type DocumentManager struct {
+	config.Options
+	securityClient clients.SecurityClient
+	fnSleeper      func(time.Duration)
+}
+
+//NewDocumentManager creates an instance or returns error
+func NewDocumentManager(config config.Options) (*DocumentManager, error) {
+	log.Tracef("Instantiating a new document manager using: %+v", config)
+	sgClient, err := clients.NewESSecurityClient(config)
+	if err != nil {
+		return nil, err
+	}
+	sleeper := func(t time.Duration) {
+		time.Sleep(t)
+	}
+	return &DocumentManager{
+		config,
+		sgClient,
+		sleeper,
+	}, nil
+}
+
+//SyncACL to include the given UserInfo
+func (dm *DocumentManager) SyncACL(userInfo *cl.UserInfo) {
+	log.Debugf("SyncACL for %+v", userInfo)
+	if dm.isInfraGroupMember(userInfo) {
+		log.Debugf("Skipping sync of ACLs for infragroup member %s. Permissions are assumed to be static", userInfo.Username)
+		return
+	}
+
+	//TODO: there must be a better way to do this.  Is Channels
+	//from small call location above better?
+	for delay := range []int{1, 1, 2, 3, 5, 8} {
+		if !dm.trySyncACL(userInfo) {
+			log.Debugf("Unable to synce ACLs, sleeping for %q seconds...", delay)
+			dm.fnSleeper(time.Duration(delay) * time.Second)
+		}
```

Comment:
else break?

Reply:
good catch... otherwise we continually trying to sync even though we succeeded